src/qtouch/qtouch.c

```diff
@@ -630,21 +629,40 @@ void qtouch_process_scroller_positions(void)
         uint8_t i, j;
         uint16_t sum = 0;
         uint16_t max_sensor_reading = 0;
+        uint16_t min_sensor_reading = DEF_SENSOR_CEILING;
         uint16_t weighted_sum = 0;
         uint16_t sensor_location[DEF_SCROLLER_NUM_CHANNELS] = {
             1, // Offset by `1` because a `0` location cannot be weight-averaged
             DEF_SCROLLER_RESOLUTION / 3,
             DEF_SCROLLER_RESOLUTION / 3 * 2,
             DEF_SCROLLER_RESOLUTION};
 
+        for (i = 0; i < DEF_SCROLLER_NUM_CHANNELS; i++) {
+            uint16_t value;
+            value = qtouch_get_sensor_node_signal_filtered(
```

Comment:
store these in a `uint16_t filtered_values[DEF_SCROLLER_NUM_CHANNELS]` array, then use the array values in the two loops below. this improves code clarity a lot, as one does not have to reason about what happens if the two loops read different values.

Reply:
ack. now done and CI passes

(used "readings" instead of "values" in order to match the naming in other variables)